docs/en/operation/Scalability-Tuning.md

```diff
@@ -8,20 +8,118 @@ priority: 9
 
 Alluxio is a scalable distributed file system designed to handle many workers within a single
 cluster.
-Several parameters can be tuned to prevent the Alluxio master from being overloaded.
-This page details the parameters to tune when scaling a cluster.
+This page details methods to recommend Alluxio node resource sizes. The section
+[Metrics to Monitor](#metrics-to-monitor) describes methods to monitor and estimate resources which
+influence the system's hardware requirements. The sections
+[Alluxio Master Configuration](#alluxio-master-configuration),
+[Alluxio Worker Configuration](#alluxio-worker-configuration), and
+[Alluxio Client Configuration](#alluxio-client-configuration) provide details on hardware and
+configuration tuning for large scale deployments.
 
 * Table of Contents
 {:toc}
 
+## Metrics To Monitor
+
+### Number of Files in Alluxio
+
+Files refers to files and directories. The number of files in Alluxio can be monitored through the
+metrics `Master.FilesCreated` plus `Master.DirectoriesCreated` minus `Master.PathsDeleted`. These
+metric will need to be aggregated by a third party metrics collector because the value will reset on
+each master restart or failover.
+
+The number of files in Alluxio impacts the following:
+* Size of heap required by the master - Each file takes approximately 1 - 2 kb. If RocksDB is used,
+most file metadata is stored off-heap, and the size of the heap impacts how many files’ metadata can
+be cached on heap. See the
+[RocksDB section]({{ 'en/operation/Metastore.html#rocksdb-metastore' | relativize_url }}) for more
+information.
+* Size of disk required for journal storage - Each file takes approximately 1 - 2 kb on disk.
+* Latency of journal replay - The journal replay, which is the majority of the cold startup time for
+a master, takes time proportional to the number of files in the system.
+* Latency of journal backup - The journal backup takes time proportional to the number of files in
+the system. If delegated backups are used, the primary master will not be impacted during the entire
+backup duration.
+* Latency of recursive operations - Recursive operations such as `loadMetadata` and `delete` take
+time proportional to the number of files in the subtree being operated on. This should not impact
+user experience unless the subtree is significantly large (> 10k files).
+
+### Number of Concurrent Clients
```

Comment:
same here, what are the actions to take if num concurrent clients is high?
shall we tune the pool size?

Reply:
The impact is detailed in each of the master/worker sections, and the methods for updating those hardware requirements are in the `Master/Worker/Client Configuration` sections. I haven't included information on how to decrease the number of concurrent clients, which is another area of workload tuning.